storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestLargeCluster.java

```diff
@@ -0,0 +1,458 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+@ExtendWith({NormalizedResourcesExtension.class})
+public class TestLargeCluster {
+    private static final Logger LOG = LoggerFactory.getLogger(TestLargeCluster.class);
+
+    public static final String TEST_CLUSTER_NAME = "largeCluster01";
+    public static final String TEST_RESOURCE_PATH = "clusterconf/" + TEST_CLUSTER_NAME;
+
+    private static IScheduler scheduler = null;
+
+    @AfterEach
+    public void cleanup() {
+        if (scheduler != null) {
+            scheduler.cleanup();
+            scheduler = null;
+        }
+    }
+
+    /**
+     * Get the list of serialized topology (*code.ser) and configuration (*conf.ser)
+     * resource files in the path. The resources are sorted so that paired topology and conf
+     * files are sequential. Unpaired files may be ignored by the caller.
+     *
+     * @param path directory in which resources exist.
+     * @return
+     * @throws IOException
+     */
+    public static List<String> getResourceFiles(String path) throws IOException {
+        List<String> fileNames = new ArrayList<>();
+
+        try (
+                InputStream in = getResourceAsStream(path);
+                BufferedReader br = new BufferedReader(new InputStreamReader(in))
+        ) {
+            String resource;
+
+            while ((resource = br.readLine()) != null) {
+                if (resource.endsWith("code.ser") || resource.endsWith("conf.ser")) {
+                    fileNames.add(path + "/" + resource);
+                }
+            }
+            Collections.sort(fileNames);
+        }
+        return fileNames;
+    }
+
+    /**
+     * InputStream to read the fully qualified resource path.
+     *
+     * @param resource
+     * @return
+     */
+    public static InputStream getResourceAsStream(String resource) {
+        final InputStream in = getContextClassLoader().getResourceAsStream(resource);
+        return in == null ? ClassLoader.getSystemClassLoader().getResourceAsStream(resource) : in;
+    }
+
+    /**
+     * Read the contents of the fully qualified resource path.
+     *
+     * @param resource
+     * @return
+     * @throws Exception
+     */
+    public static byte[] getResourceAsBytes(String resource) throws Exception {
+        InputStream in = getResourceAsStream(resource);
+        if (in == null) {
+            return null;
+        }
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+            while (in.available() > 0) {
+                out.write(in.read());
+            }
+            return out.toByteArray();
```

Comment:
I was wondering if `return` within the try-with-resources block will close the resources or not. And I found this https://stackoverflow.com/questions/22947755/try-with-resources-and-return-statements-in-java/22947904

So it should be fine.

Reply:
Yes. Pretty cool JDK 8 feature. Here is more authoritative info - which actually uses return as an example too: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
Agreed - try with resources will close as part of the "destructor"